src/Symfony/Component/HttpKernel/Controller/ControllerReference.php

```diff
@@ -26,19 +26,15 @@
  */
 class ControllerReference
 {
-    public string $controller;
-    public array $attributes = [];
-    public array $query = [];
```

Comment:
Thinking to the property initialization issue. This class could be final.

Reply:
I've reverted the two properties that have an initializer. I'm not 100% happy with that, but maybe I can find a better solution in a follow-up. 🤔